### PR TITLE
Rebalance pp awarded for flashlight plays

### PR DESCRIPTION
**Pull request by MBmasher**
As proposed in [#44](https://github.com/ppy/osu-performance/issues/44).

Nerfs short maps, buffs long maps.

### CODE REVIEW

**Comment by MBmasher** [comment]
The only aspect of difficulty that combo affects is the 100x and 200x combo thing, but it's mostly just objects, and buzz sliders could become overweighted.

**Comment by Dawnsday** [comment]
This PR is so unbelievably dead.

Anyway a followup thought I had, thoughts on stream maps HDFL? It may be worth experimenting with different formulas since I could see stream maps still coming out of this being underweighted but thats just a thought I'm spitballing

**Comment by MBmasher** [comment]
Well, we could probably buff speed if HD and FL are used together since HDFL streams are near impossible. I'll try to figure out a formula.

File: Src/Processor/Standard/StandardScore.cpp [hunk 1]
```diff
@@ -127,8 +127,10 @@ void StandardScore::computeAimValue(const Beatmap& beatmap)
 		_aimValue *= 1.03f;
 
 	if ((_mods & EMods::Flashlight) > 0)
-		// Apply length bonus again if flashlight is on simply because it becomes a lot harder on longer maps.
-		_aimValue *= 1.45f * LengthBonus;
+		// Apply object-based bonus for flashlight.
+		_aimValue *= 1.0f + 0.35f * std::min(1.0f, static_cast<f32>(numTotalHits) / 250.0f) +
++        		(numTotalHits > 250 ? 0.3f * std::min(1.0f, static_cast<f32>(numTotalHits - 250) / 250.0f) +
++        		(numTotalHits > 500 ? static_cast<f32>(numTotalHits - 500) / 1200.0f : 0.0f) : 0.0f);
```
Comment (cl8n) [inline Src/Processor/Standard/StandardScore.cpp]: what's up with the leading plus signs on these lines, looks like you pasted from a diff or something

**Comment by MBmasher** [comment]
yeah fixed

**Comment by Tom94** [comment]
Can you run this on the latest database dumps (see README) and post the top-100 flashlight plays in terms of pp? Just to make sure there are no crazy outliers

**Comment by qewc** [comment]
I can when I get home, if you'd want @Tom94

**Comment by peppy** [comment]
poke @smoogipoo

**Comment by smoogipoo** [comment]
Reference: http://original.smgi.me/rankings/osu/performance
This PR: http://mbmasher-fl.smgi.me/rankings/osu/performance

(You'll likely find most top-100 FL plays in the top 10k players)

**Comment by Tom94** [comment]
Once again, I do not believe this is an adequate way to gauge the effect of this change. We need a list of top-100 _scores_. Nobody has the time to manually go through all the players with differing pp, manually finding which scores were done with FL, and then hoping those actually correspond to the previously broken ones.

**Comment by peppy** [comment]
Intention was to pull the top 100 from the database after running. We'll make this happen.

**Comment by smoogipoo** [comment]
Here’s the data for top 100 FL scores:

Reference: https://hastebin.com/gozinuxosu.rb
mbmasher-fl: https://hastebin.com/luguyubenu.rb

**Comment by Tom94** [comment]
That's already a lot better, but is there any chance to sort the second list the same way as the first, to diff the pp values (e.g. +123 or -5), and to display the map name + mods (especially to see whether HD, DT and HR also were present without manually interpreting bits) alongside the remaining stats?

**Comment by Tom94** [comment]
We should have that kind of diff probably set up as a script that we can run on _any_ change, not just for this particular one. I can not think of a scenario where we would _not_ want this very basic of verification.

**Comment by smoogipoo** [comment]
In the future we should have a better system for this. I'm far too busy right now to work on this, but I have tabulated the data in what I hope is a readable format:

https://docs.google.com/spreadsheets/d/14vb_5bRbPxwMiiLpsU-YxkRniR27ODg6mC68Ae1inz8/edit#gid=870463036&fvid=607160840

**Comment by Tom94** [comment]
That's perfect, thanks! :) Yeah, we need an automated way to generate tables like this.

**Pull request merged**